core/src/main/java/org/infinispan/distribution/ch/impl/DefaultConsistentHash.java

```diff
@@ -205,15 +205,14 @@ public String toString() {
    @Override
    public String getRoutingTableAsString() {
       StringBuilder sb = new StringBuilder();
-      for (Address a : members) {
-         if (sb.length() > 0) {
-            sb.append("\n  ");
+      for (int i = 0; i < segmentOwners.length; i++) {
+         if (i > 0) {
+            sb.append(", ");
+         }
+         sb.append(i).append(':');
+         for (int j = 0; j < segmentOwners[i].size(); j++) {
+            sb.append(' ').append(members.indexOf(segmentOwners[i].get(j)));
```

Comment:
don't you want to append the `Address` instead of the index?

Reply:
No, I put the index because this layout repeats the owner (index) a lot. Especially in the test suite, with the node name being `VeryLongTestName-NodeA-12345`, logging the index instead of the name saves a lot.

Logging the node name works ok with the previous layout, which logs each node name once and then each segment id numOwner times, but I found that in some rare cases I care about the order too. I guess I could have modified the layout to print primary owned segments first, then segments where we are the 1st backup, then segments where we are the 2nd backup etc, but going back to the previous layout seemed simpler.